inst/rmd/ioslides/ioslides-13.5.1/js/slide-deck.js

```diff
@@ -793,17 +793,20 @@ var loadDeck = function(event) {
   });
 };
 
-if (document.readyState !== "loading" &&
-    document.querySelector('slides') === null) { 
-  // if the document is done loading but our element hasn't yet appeared, defer
-  // loading of the deck
-  window.setTimeout(function() {
-    loadDeck(null);
-  }, 0);
-} else {
-  // still loading the DOM, so wait until it's finished
-  document.addEventListener("DOMContentLoaded", loadDeck);
+if (!window.Shiny) {
+  // If Shiny is loaded, the slide deck is initialized in ioslides template
+
+  if (document.readyState !== "loading" &&
+      document.querySelector('slides') === null) {
+    // if the document is done loading but our element hasn't yet appeared, defer
+    // loading of the deck
+    window.setTimeout(function() {
+      loadDeck(null);
+    }, 0);
```

Comment:
Here we are deferring if no `slides` which happens with new async behavior 

So we could also adapt that part to correctly wait for this element, right ?

Reply:
It seems when Shiny is on the page is the problem here. `slide-deck.js` is run before `windows.Shiny` exists it seems. 

So we could do what you say by moving the whole `loadDeck()` call outside of `slide-deck.js` to be called inside the pandoc template that will be inserted as UI. 

I don't know if there is a way to make `slide-deck.js` loads after Shiny.

Reply:
It's not worth the complexity. Technically, you could have been calling `loadDeck()` in the ioslides pandoc template this whole time. But I'd still keep this code here for portability reasons.

Similarly, yes there's lots of refactoring that could happen in this code but I don't recommend making any changes right now.

@cpsievert The Shiny task queue isn't a good fit here for a few of reasons. But primarily this file doesn't use jQuery and I don't think it should use Shiny methods either, beyond just testing for its presence.

Reply:
Ok great ! 

Thanks for your feedback. All good - I'll accept this change.